<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>EDUCE FAQ</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">EDUCE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Course Info
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://www.google.com/?gws_rd=ssl">Course 1 (Add HTML version of Rmd in href)</a>
    </li>
    <li>
      <a href="https://www.google.com/?gws_rd=ssl">Course 2 (Add HTML version of Rmd in href)</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://ubc-educe.shinyapps.io/course_knitter/">Course Compiler</a>
</li>
<li>
  <a href="https://ubc-educe.shinyapps.io/data_manipulator/">Data Manipulator</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">EDUCE FAQ</h1>

</div>


<div id="what-is-educe" class="section level1">
<h1>What is EDUCE?</h1>
<p>Experiential Data Science for Undergraduate Cross-disciplinary Education (EDUCE)</p>
<p>EDUCE seeks to develop a uniform experiential learning framework that is cross-disciplinary and collaborative to equip undergraduate students in the life sciences with basic competency and literacy in data science as it pertains to microbiome sequence information. To achieve this, EDUCE will develop and implement a mechanism to augment undergraduate training in the Department of Microbiology &amp; Immunology (MICB) with basic analyses and statistical principles and microbiome data processing and interpretation skills that transcend any one course in the department.</p>
</div>
<div id="what-is-course-compiler" class="section level1">
<h1>What is Course Compiler?</h1>
<p>Here you can select and compile all the data science modules for either teaching purposes and/or self-learning purposes. Select the courses that best fit your interests/curriculum and have the Course Compiler Shiny App knit it for you. We will also provide the source code for generating the arbitrary knitted files and enable you to customize courses with your own files. More experienced data science instructors will have access to more functions, so we encourage you to learn the basics of RStudio.</p>
<p>For better performance, veteran instructors can download the app and run it locally on their machine (Requires RStudio):</p>
<ol style="list-style-type: decimal">
<li><p>Go to the <a href="https://github.com/EDUCE-UBC/EDUCE-UBC.github.io/tree/master/Downloadable_Apps">Course Compiler</a> app folder in our public repo on Github.</p></li>
<li><p>Copy the link and paste it at <a href="https://kinolien.github.io/gitzip/">Gitzip</a> (Top right corner beside the key icon) to download.</p></li>
<li><p>Open the app.R file in RStudio.</p></li>
<li><p>Click the “Run App” button or type “shiny::runApp()” in the console to launch the app.</p></li>
</ol>
<div id="what-course-compiler-offers" class="section level3">
<h3>What Course Compiler Offers</h3>
<ul>
<li>Ability to select and knit courses that interest you and/or fit your teaching plans.</li>
<li>Ability to select the HTML format for novice data science instructors or the ZIP format for more experienced data science instructors (Source code will be provided in the ZIP format for you to use and personalize your own courses).</li>
<li>More experienced data science instructors will have access to knitted files that contain a table of content.</li>
</ul>
</div>
<div id="provided-files-in-zip-format" class="section level3">
<h3>Provided Files in ZIP Format</h3>
<ul>
<li>Rmd_Output = All the courses that you selected.</li>
<li>Rmd_Master_File = Master file containing R children for all the knitted courses.</li>
<li>User_File = Source code used to generate all the courses that you want.</li>
</ul>
</div>
<div id="how-do-i-create-courses-with-my-own-rmd-files" class="section level3">
<h3>How do I create courses with my own Rmd files?</h3>
<p>To create courses with your own Rmd files:</p>
<ol style="list-style-type: decimal">
<li>Select the courses offered by EDUCE in Course Compiler.</li>
<li>Select the ZIP format for download.</li>
<li>After downloading, extract the files/folders.</li>
<li>Put all the files that you want to knit into the folder “Rmd_Output”.</li>
<li>Knit the “User_File.Rmd”, which will generate a “Rmd_Master_File.Rmd” that contains all the R children of the files.</li>
<li>Finally, knit the “Rmd_Master_File.Rmd” into HTML or PDF.</li>
</ol>
</div>
</div>
<div id="what-is-data-manipulator" class="section level1">
<h1>What is Data Manipulator?</h1>
<p>The Data Manipulator Shiny App has two functionalities:</p>
<ul>
<li>Provide a user-friendly way to create data plots given an input file.</li>
<li>Provide an interactive way to learn the various methods of normalization in a tabular and graphical way.
<ul>
<li>Users will be able to use either OTU/ASV count files as inputs for normalization.</li>
<li>Users will be able to normalize the data via PRA/RS, view the most abundant OTU/ASV, and download a copy of the normalized data as text files.</li>
</ul></li>
</ul>
<p>Novice data science instructors will be provided some basic files from Github to experiment with and more experienced data science instructors will have the ability to use their own files to do some more in-depth experimentation and manipulation of data via the graph inputs (Number of OTU/ASV to Display, Number of Columns to Display, and Random Generator Seed). The Shiny app will also allow users to download the normalized data as text files (Note: Only Percent Relative Abundance and Random Subsampling are in the app due to computation limitations).</p>
<p>For better performance, veteran instructors can download the app and run it locally on their machine (Requires RStudio):</p>
<ol style="list-style-type: decimal">
<li><p>Go to the <a href="https://github.com/EDUCE-UBC/EDUCE-UBC.github.io/tree/master/Downloadable_Apps">Data Manipulator</a> app folder in our public repo on Github.</p></li>
<li><p>Copy the link and paste it at <a href="https://kinolien.github.io/gitzip/">Gitzip</a> (Top right corner beside the key icon) to download.</p></li>
<li><p>Open the app.R file in RStudio.</p></li>
<li><p>Click the “Run App” button or type “shiny::runApp()” in the console to launch the app.</p></li>
</ol>
<div id="file-inputs-for-data-manipulator" class="section level3">
<h3>File Inputs For Data Manipulator</h3>
<ul>
<li>Data Plotter:
<ul>
<li>Metadata (Txt File)</li>
</ul></li>
<li>Data Normalization:
<ul>
<li>OTU Count (Txt File)</li>
<li>ASV Count (Txt File)</li>
</ul></li>
</ul>
</div>
<div id="purpose-of-validate-data" class="section level3">
<h3>Purpose of “Validate Data”?</h3>
<p>More experienced data science instructors may help us test the Data Normalization section of the Shiny app version (Random Subsampling Section) by providing us with their OTU/ASV files for integration testing. To submit your file(s), upload your file and press the “Validate Data” button. Assuming the files are valid and in the correct format, this will save a copy of your file on our server and we may use it for future tests.</p>
</div>
</div>
<div id="data-normalization" class="section level1">
<h1>Data Normalization</h1>
<p>Data normalization is the process of transforming/standardizing data to a common scale for comparison [4]. This is especially useful for microbial ecology since the data may come from different samples and a way to compare the trends of data on a standardized scale will be useful for further analysis [5].</p>
<div id="percent-relative-abundance" class="section level2">
<h2>Percent Relative Abundance</h2>
<p>Percent Relative Abundance is a technique that transforms the data into percentages, which is then used to compare how the data is relative to one another. Also known as Relative Species Abundance in microbial ecology, it is a measure of how common a species is relative to other species in a defined location [6].</p>
<p>This technique first identifies the total number of observations represented in each group, then each data point is divided by the total number of observations calculated earlier. The sum of all the percentages in each group should equal 100% [7].</p>
<div id="potential-benefits" class="section level4">
<h4>Potential Benefits:</h4>
<ul>
<li>Can easily conceptualize data without overthinking [8]: Dealing with percentages vs. actual numbers without context.</li>
<li>Easy to transform data:
<ul>
<li>[Number of observations for each data point] / [Total number of observations for each group of data points]</li>
</ul></li>
</ul>
</div>
<div id="potential-drawbacks" class="section level4">
<h4>Potential Drawbacks:</h4>
<ul>
<li>Difficult to comprehend complex datasets [8]. This can potentially be resolved by dividing large datasets into interval classes for better visualization.</li>
</ul>
</div>
</div>
<div id="random-subsampling" class="section level2">
<h2>Random Subsampling</h2>
<p>Random Subsampling is technique that splits the data into subsets [9]. Also known as rarefaction in microbial ecology, it is a technique used to determine species richness of samples that differ in area, volume, or sampling efforts [10].</p>
<div id="potential-benefits-1" class="section level4">
<h4>Potential Benefits:</h4>
<ul>
<li>Subsetting of samples can be repeated an indefinite number of times.</li>
<li>Rarefaction compares observed richness among samples for a given level of sampling effort and does not attempt to estimate true richness of community [11]: Number of observations for each subsampled data point Total number of observations for each group of subsampled data points</li>
</ul>
</div>
<div id="potential-drawbacks-1" class="section level4">
<h4>Potential Drawbacks:</h4>
<ul>
<li>Rarefied counts remain over-dispersed relative to Poisson model (Implies an increase in Type I error). More difficult to estimate overdispersion after rarefying due to lost information [12].</li>
<li>Rarefied counts represent only a small fraction of original data (Can lead to increased chance of Type II error). Often caused by samples being discarded and poorly distinguishable samples due to discarded fraction of original library [12].</li>
<li>Random step in rarefying is unnecessary and adds artificial uncertainty [12].</li>
<li>Many assumptions are required if rarefaction is used for comparing samples: Sufficient sampling, comparable sampling methods, taxonomic similarity, closed communities of discrete individuals, random placement, and independent random sampling [13, 14].</li>
</ul>
</div>
</div>
<div id="multiple-imputation" class="section level2">
<h2>Multiple Imputation</h2>
<p>Multiple Imputation is a statistical technique that is useful for analyzing incomplete or missing data via 3 steps [15, 16]:</p>
<ol style="list-style-type: decimal">
<li>Imputation: Missing entries of datasets are filled in m times, which can contain different values for each missing entry (m Complete datasets).</li>
<li>Analysis: The m completed datasets are then analyzed (m Analyses).</li>
<li>Pooling: The m analysis results are then pooled together into a final result. The MICE Package was used in R with Predictive Mean Matching (PMM) as the imputation method (A list of imputation methods can also be found here) [17].</li>
</ol>
<div id="potential-benefits-2" class="section level4">
<h4>Potential Benefits:</h4>
<ul>
<li>When Multiple Imputation is used correctly [18, 19, 20, 21, 22]:
<ul>
<li>Reduced bias: Analysis can be done on a “complete” dataset via filling missing values with imputation methods.</li>
<li>Improve validity: Valid inferences are obtained from the distribution average of missing data.</li>
<li>Increase precision: Many missing data, which can potentially happen in microbial ecology due to the potential of excess zeros being missing data, may lead to the exclusion of a substantial proportion of the original sample, leading to a decrease in precision during analysis.</li>
<li>Result in robust statistics (Resistant to outliers): Imputed values often uses the mean, median, or other statistic to impute the missing values.</li>
</ul></li>
<li>Useful for analyzing qPCR data, which often have moderate levels of missing values [20].</li>
</ul>
</div>
<div id="potential-drawbacks-2" class="section level4">
<h4>Potential Drawbacks:</h4>
<ul>
<li>Must make a few required statistical assumptions [23]:
<ul>
<li>Is the data is missing at random?</li>
<li>Is this a multivariate normal distribution? (Needed for some of the modeling methods)</li>
</ul></li>
<li>Need to transform variables to approximate normal distribution before running imputation procedure if data isn’t normally distributed [18].</li>
<li>Incorrect model choices or exclusion of vital data points may lead to more bias [18].</li>
</ul>
</div>
</div>
<div id="variance-stabilizing-transformation" class="section level2">
<h2>Variance Stabilizing Transformation</h2>
<p>Variance Stabilizing Transformation is a technique that uses a function f to apply values to x in a dataset to create y = f(x) such that the variability of values y is not related to their mean value (Or has a constant variance) [24].</p>
<div id="potential-benefits-3" class="section level4">
<h4>Potential Benefits:</h4>
<ul>
<li>Usage of DESeq2 overcomes the issue of strong variance of logarithmic fold change (LFC) estimates for genes with low read count in analysis of high-throughput sequencing data [25]:</li>
<li>Shrinks LFC estimates toward zero when shrinkage is stronger when available information or gene is low, which could be due to low counts, high dispersion, or few degrees of freedom.</li>
<li>Prevents the spreading apart of data for genes with low counts, where random noise is likely to dominate any biologically meaningful signal (Stabilizes variances).</li>
<li>DeSeq2’s heuristics can deal with flagged outliers in two possible ways to help avoid Type I errors [25]:</li>
<li>Outliers in conditions with six or fewer replicates cause whole gene to be flagged and removed from subsequent analysis (Includes P value adjustment for multiple testing).</li>
<li>Outliers in conditions with seven or more replicates replaces the outlier counts with imputed value (Trimmed mean over all samples), scaled by size factor, and then re-estimates dispersion, LFCs, and P values for these genes.</li>
<li>Can consistently perform over large range of data types and is applicable for small studies with few replicates or large observational studies [25].</li>
</ul>
</div>
<div id="potential-drawbacks-3" class="section level4">
<h4>Potential Drawbacks:</h4>
<ul>
<li>Many rare species are ignored completely due to the negative values from log-like transformations being set to 0. Since DESeq was developed mainly for use with Euclidean metrics (Negative results are not a problem), this may yield misleading results for ecologically useful non-Euclidean measures [26].</li>
<li>Many not be appropriate for highly diverse microbial environments since DESeq assumes that most microbes are not differentially abundance and for the few that are, there is an approximately balanced amount of increased/decreased abundance [26].</li>
</ul>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>[1] Holmes R. Software Testing. <a href="https://github.com/ubccpsc/310/blob/2017jan/readings/Testing.md" class="uri">https://github.com/ubccpsc/310/blob/2017jan/readings/Testing.md</a> (2018).</p>
<p>[2] Anonymous. Big Bang Integration Testing. <a href="http://www.professionalqa.com/big-bang-integration-testing" class="uri">http://www.professionalqa.com/big-bang-integration-testing</a> (2018).</p>
<p>[3] Easterbrook S. Testing Strategies. <a href="http://www.cs.toronto.edu/~sme/CSC302/notes/17-Testing2.pdf" class="uri">http://www.cs.toronto.edu/~sme/CSC302/notes/17-Testing2.pdf</a> (2018).</p>
<p>[4] Borgatti S. <a href="http://www.analytictech.com/ba762/handouts/normalization.htm" class="uri">http://www.analytictech.com/ba762/handouts/normalization.htm</a> (2018).</p>
<p>[5] Daniel Aguirre de Cárcer, Denman SE, McSweeney C, Morrison M. Evaluation of Subsampling-Based Normalization Strategies for Tagged High-Throughput Sequencing Data Sets from Gut Microbiomes. Applied and Environmental Microbiology. 2011; 77: 8795-8798.</p>
<p>[6] Socratic. How do species richness and relative abundance of species affect species diversity? <a href="https://socratic.org/questions/how-do-species-richness-and-relative-abundance-of-species-affect-species-diversi" class="uri">https://socratic.org/questions/how-do-species-richness-and-relative-abundance-of-species-affect-species-diversi</a> (2018).</p>
<p>[7] Percentage Frequency Distribution. Encyclopedia of Survey Research Methods. 2008.</p>
<p>[8] Reid A. Advantages &amp; Disadvantages of a Frequency Table. <a href="https://sciencing.com/advantages-disadvantages-frequency-table-12000027.html" class="uri">https://sciencing.com/advantages-disadvantages-frequency-table-12000027.html</a> (2018).</p>
<p>[9] Dieterle F. Random Subsampling. <a href="http://www.frank-dieterle.com/phd/2_4_3.html" class="uri">http://www.frank-dieterle.com/phd/2_4_3.html</a> (2018).</p>
<p>[10] Chiarucci A, Bacaro G, Rocchini D, Ricotta C, Palmer MW, Scheiner SM. Spatially constrained rarefaction: incorporating the autocorrelated structure of biological communities into sample-based rarefaction. Community Ecology. 2009; 10: 209-214.</p>
<p>[11] Hughes JB, Hellmann JJ. The application of rarefaction techniques to molecular inventories of microbial diversity. In: Vol 397. United States: Elsevier Science &amp; Technology; 2005: 292-308.</p>
<p>[12] McMurdie PJ, Holmes S. Waste not, want not: why rarefying microbiome data is inadmissible. PLoS Computational Biology. 2014; 2013; 10: e1003531.</p>
<p>[13] Gotelli, NJ, Colwell RK. Estimating species richness. Frontiers in Measuring Biodiversity. 2011; 12: 39-54.</p>
<p>[14] Tipper JC. Rarefaction and Rarefiction; The Use and Abuse of a Method in Paleoecology. Paleobiology. 1979; 5: 423-434.</p>
<p>[15] van Buuren S. Multiple Imputation. <a href="http://www.stefvanbuuren.nl/mi/mi.html" class="uri">http://www.stefvanbuuren.nl/mi/mi.html</a> (2018).</p>
<p>[16] Maldonado, A. D.; Aguilera, P. A.; and Salmeron, A. An Experimental Comparison of Methods to Handle Missing Values in Environmental Datasets. International Congress on Environmental Modelling and Software. 2016: 3.</p>
<p>[17] van Buuren S. MICE. <a href="https://www.rdocumentation.org/packages/mice/versions/2.25/topics/mice" class="uri">https://www.rdocumentation.org/packages/mice/versions/2.25/topics/mice</a> (2018).</p>
<p>[18] Anonymous. Statistics How To. <a href="http://www.statisticshowto.com/multiple-imputation/" class="uri">http://www.statisticshowto.com/multiple-imputation/</a> (2018).</p>
<p>[19] Sterne JAC, White IR, Carlin JB, et al. Multiple imputation for missing data in epidemiological and clinical research: potential and pitfalls. BMJ. 2009; 338: 157-160.</p>
<p>[20] Kanwar N, Scott HM, Norby B, et al. Impact of treatment strategies on cephalosporin and tetracycline resistance gene quantities in the bovine fecal metagenome. Scientific Reports. 2014; 2015; 4: 5100.</p>
<p>[21] Xu L, Paterson AD, Turpin W, Xu W. Assessment and Selection of Competing Models for Zero-Inflated Microbiome Data. PLoS One. 2015; 10: e0129606.</p>
<p>[22] Kaul A, Mandal S, Davidov O, Peddada SD. Analysis of Microbiome Data in the Presence of Excess Zeros. Frontiers in Microbiology. 2017; 8: 2114.</p>
<p>[23] Quora. How do you handle missing data (statistics)? What imputation techniques do you recommend or follow? <a href="https://www.quora.com/How-do-you-handle-missing-data-statistics-What-imputation-techniques-do-you-recommend-or-follow" class="uri">https://www.quora.com/How-do-you-handle-missing-data-statistics-What-imputation-techniques-do-you-recommend-or-follow</a> (2018).</p>
<p>[24] NC State University. Nonlinear Statistical Models for Univariate and Multivariate Response. <a href="https://www.stat.ncsu.edu/people/bloomfield/courses/ST762/slides/MD-02-2.pdf" class="uri">https://www.stat.ncsu.edu/people/bloomfield/courses/ST762/slides/MD-02-2.pdf</a> (2018).</p>
<p>[25] Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome biology. 2014; 15: 550-550.</p>
<p>[26] Weiss S, Xu ZZ, Peddada S, et al. Normalization and microbial differential abundance strategies depend upon data characteristics. Microbiome. 2017; 5: 27.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
